org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/desktop/DashboardController.java

```diff
@@ -581,8 +590,14 @@ else if (btn.getAttribute("AD_Process_ID") != null)
             	{
             		int processId = (Integer)btn.getAttribute("AD_Process_ID");
             		String parameters = (String)btn.getAttribute("ProcessParameters");
-            		if (processId > 0)
-            			openReportInViewer(processId, parameters);
+            		if (processId > 0) {
+            			if (btn.getAttribute("Refresh") != null) {
+            				Component parent = (Component) btn.getAttribute("Parent");
+            				embedReport(parent, processId, parameters, true);
+            			}
+            			else
+            				openReportInViewer(processId, parameters);
+            		}
```

Comment:
this change is not needed if you use the btn.addEventListener changes suggested above.